lib/apiservers/engine/backends/system.go

```diff
@@ -240,7 +241,8 @@ func (s *System) UnsubscribeFromEvents(chan interface{}) {
 }
 
 func (s *System) AuthenticateToRegistry(ctx context.Context, authConfig *types.AuthConfig) (string, string, error) {
-	return "", "", fmt.Errorf("%s does not implement System.AuthenticateToRegistry", ProductName())
+	defer trace.End(trace.Begin(""))
+	return RegistryService.Auth(authConfig, dockerversion.DockerUserAgent(ctx))
```

Comment:
@hickeng I looked into using the docker/distribution/registry/client directly.  There isn't a login call.  The registry client in distribution provides a rountripper to add to an http client that will handle the authentication and authorization, but it doesn't actually provide a function that does the authentication.  The docker/docker/registry/auth is effectively the login client.  If we don't want to import docker/docker and import docker/distribution only for login, we would basically duplicate the loginV2() function in docker/docker/registry/auth.go.

RegistryService.Auth() that we use in our code doesn't just auth and differentiate between V1 or V2.  It handles the login option of ip:port or cidr format.  It also handles the insecure registry and does some work with a mirror registry option.

If we want to avoid importing code, we'll have to reimplement the same code.  If all we want to do is to avoid V1, we can set V2Only to true when creating the RegistryService.

Lastly, we can also import some of the imagec code and modify it to do what we need.

At this point, I think we should leave everything as is, with the exception of disabling the V1 registries.